service/controller/resource/monitoring/verticalpodautoscaler/resource.go

```diff
@@ -212,14 +212,10 @@ func (r *Resource) getMaxMemory(ctx context.Context) (*resource.Quantity, error)
 }
 
 func quantityMultiply(q *resource.Quantity, multiplier float64) (*resource.Quantity, error) {
-	i, ok := q.AsInt64()
-	if !ok {
-		return nil, microerror.Maskf(quantityConversionError, q.String())
-	}
-
-	n := float64(i) * multiplier
 
-	q.Set(int64(n))
+	i := q.AsApproximateFloat64()
+	n := i * multiplier
+	q.SetMilli(int64(n * 1000))
```

Comment:
Why use `SetMilli` to then multiply it by 1000 ?

Reply:
We have a number like `2.5`
`Set` requires cast to int => becomes `2`
and `SetMilli` uses milli-cpus, so expects `2500`.

Reply:
:ack: